juju/client/facade.py

```diff
@@ -675,21 +675,43 @@ def from_json(cls, data):
                 raise
         if isinstance(data, list):
             # check: https://juju.is/docs/sdk/assumes
-            # assumes are in the form of a list
             d = {}
             for entry in data:
-                if '>' in entry or '>=' in entry:
-                    # something like juju >= 2.9.31
-                    i = entry.index('>')
-                    key = entry[:i].strip()
-                    value = entry[i:].strip()
-                    d[key] = value
+                if isinstance(entry, dict):
+                    # this could be
+                    # any-of
+                    # all-of
+                    for _, v in entry.items():
+                        if isinstance(v, list):
+                            for i in v:
+                                cls.splitEntries(i, d)
+                        else:
+                            cls.splitEntries(v, d)
```

Comment:
I think having a static method mutating the input in a complex control like this will give us long debugging sessions in the future :) my suggestion would be to avoid side-effects in this part of the code (I mean in the `facade.py` where we generate client codes).